src/settings.cpp

```diff
@@ -426,6 +426,32 @@ s16 Settings::getS16(const std::string &name) const
 }
 
 
+v3s16 Settings::getV3S16(const std::string &name) const
+{
+	v3s16 value;
+	Strfnd f(get(name));
+	f.next("(");
+	value.X = stoi(f.next(","), -32768, 32767);
+	value.Y = stoi(f.next(","), -32768, 32767);
+	value.Z = stoi(f.next(")"), -32768, 32767);
+	return value;
+}
+
+
+v3s16 Settings::getMapGenerationLimit() const
+{
+	v3s16 mgl;
+	if (get("map_generation_limit").find("(")==std::string::npos){
+			mgl.X = getS16("map_generation_limit");
+			mgl.Y = mgl.X;
+			mgl.Z = mgl.X;
+			return mgl;
+	}
+	mgl = getV3S16("map_generation_limit");
```

Comment:
Maybe it would be a better idea to get the map_generation_limit string as raw and then try parsing it both ways, separate from settings.